runtime/serde/serde-xml/common/src/aws/smithy/kotlin/runtime/serde/xml/tokenization/XmlLexer.kt

```diff
@@ -8,46 +8,6 @@ import aws.smithy.kotlin.runtime.serde.DeserializationException
 import aws.smithy.kotlin.runtime.serde.xml.XmlToken
 import aws.smithy.kotlin.runtime.util.text.codePointToChars
 
-// https://www.w3.org/TR/xml/#NT-S
-private val xmlWhitespaceChars = setOf(
-    '\t', // horizontal tab
-    '\r', // carriage return
-    '\n', // newline
-    ' ', // normal space
-)
-private fun Char.isXmlWhitespace() = xmlWhitespaceChars.contains(this)
-
-private fun Char.toRange() = this..this
-
-// https://www.w3.org/TR/xml/#NT-Name
-private val nameStartCharRanges = setOf(
```

Comment:
Agreed, it was more readable before. I think the bottleneck is because neither the JVM stdlib nor Kotlin stdlib special-cases small set sizes nor scalar sets. This used a full linked hashset implementation with Char boxing/unboxing to perform the lookups, which was slower than a tight `if` condition directly on scalars.